yarn-project/key-store/src/test_key_store.ts

```diff
@@ -151,22 +153,27 @@ export class TestKeyStore implements KeyStore {
   }
 
   /**
-   * Derives and returns the application nullifier secret key for a given account or master nullifier public key hash.
-   * @throws If the account does not exist in the key store.
-   * @param accountOrNpkMHash - account address or master nullifier public key hash.
+   * Derives and returns the application nullifier secret key for a given master nullifier public key hash.
+   * @throws If the account corresponding to the master nullifier public key hash does not exist in the key store.
+   * @param npkMHash - The master nullifier public key hash.
    * @param app - The application address to retrieve the nullifier secret key for.
    * @returns A Promise that resolves to the application nullifier secret key.
    */
-  public async getAppNullifierSecretKey(accountOrNpkMHash: AztecAddress | Fr, app: AztecAddress): Promise<Fr> {
+  public async getAppNullifierSecretKey(npkMHash: Fr, app: AztecAddress): Promise<Fr> {
+    const { key } = this.#getMapMetadataForMasterNullifierPublicKeyHash(npkMHash);
```

Comment:
You are right. Was not that great. I refactored it a bit [here](https://github.com/AztecProtocol/aztec-packages/pull/6405/commits/6989d9f4cc00d5319e7cfa9176c25930dfc960c3) so that it's a bit nicer. It's still too much code for too little functionality but that's probably just inevitable with the key rotation and us having to deal with npk_m_hash and not just npk_m.